R/data_read.R

```diff
@@ -42,35 +47,51 @@
 #' @section Differences to other packages that read foreign data formats:
 #' `data_read()` is most comparable to `rio::import()`. For data files from
 #' SPSS, SAS or Stata, which support labelled data, variables are converted into
-#' their most appropriate type. The major difference to `rio::import()` is
-#' that `data_read()` automatically converts variables into factors, unless
-#' the variables are only partially labelled, in which case variables are
-#' converted to numerics. Character vectors are preserved. Hence, variables,
-#' where _all_ values are labelled, will be converted into factors, where
-#' imported value labels will be set as factor levels. Else, if a variable
-#' has _no_ value labels or less value labels than values, the variable is
-#' either converted into numeric or character vector. Value labels are then
-#' preserved as `"labels"` attribute.
+#' their most appropriate type. The major difference to `rio::import()` is that
+#' `data_read()` automatically converts fully labelled numeric variables into
+#' factors, where imported value labels will be set as factor levels. If a
+#' numeric variable has _no_ value labels or less value labels than values, it
+#' is not converted to factor. In this case, value labels are preserved as
+#' `"labels"` attribute. Character vectors are preserved.  Use
+#' `convert_factors = FALSE` to remove the automatic conversion of numeric
+#' variables to factors.
 #'
 #' @export
-data_read <- function(path, path_catalog = NULL, encoding = NULL, verbose = TRUE, ...) {
+data_read <- function(path,
+                      path_catalog = NULL,
+                      encoding = NULL,
+                      convert_factors = TRUE,
+                      verbose = TRUE,
+                      ...) {
   # extract first valid file from zip-file
   if (.file_ext(path) == "zip") {
     path <- .extract_zip(path)
   }
 
   # read data
-  switch(.file_ext(path),
+  out <- switch(.file_ext(path),
     "txt" = ,
     "csv" = .read_text(path, encoding, verbose, ...),
     "xls" = ,
     "xlsx" = .read_excel(path, encoding, verbose, ...),
     "sav" = ,
```

Comment:
@strengejacke can R read `sav` files? If not we can use `haven::read_sav()`

Reply:
No, we need (and rely on) _haven_ to do this (alternative would be the _foreign_ package). Should also be documented in the help file which packages are used.

Reply:
Yes but I don't understand why you specify e.g `.read_spss` for `por` files and you don't specify `.read_sav` for `sav` files

Reply:
Ok I think I got it, "sav" also redirects to `.read_spss()`. I'm not used to `switch()`, my bad.